TestFramework/Tests.Integration/Tests/DataStoreNetworkIntegrationTests.cs

```diff
@@ -1517,6 +1517,144 @@ where todo.Name.StartsWith("a to") || todo.Details.Equals("details for 2")
 		}
 
 		[Test]
+		public async Task TestNetworkStoreFindByQueryMultipleWhereClausesEquals()
```

Comment:
@vinaygahlawat this test makes sense. As a next step, we really should have "unit" tests on the query module that just tests the query conversion logic rather than going black-box from the datastore.